src/coord/rmr/rmr.c

```diff
@@ -503,11 +510,7 @@ void MRIteratorCallback_Done(MRIteratorCallbackCtx *ctx, int error) {
   // Mark the command of the context as depleted (so we won't send another command to the shard)
   ctx->cmd.depleted = true;
   short pending = --ctx->it->ctx.pending; // Decrease `pending` before decreasing `inProcess`
-  if (pending <= 0) {
-    RS_LOG_ASSERT(pending >= 0, "Pending should not reach a negative value");
-    // Unblock the channel before calling `ProcessDone`, as it may trigger the freeing of the iterator
-    MRChannel_Unblock(ctx->it->ctx.chan);
-  }
+  RS_LOG_ASSERT(pending >= 0, "Pending should not reach a negative value");
```

Comment:
Can we replace this with a simple `assert` call?

Reply:
what's the diffrence?

Reply:
Hopefully, soon, we will have dev-time and prod-time assertions, this one should be a dev-time assertion IMO, so no need for a redis module API assertion

Reply:
why do you think this is a dev time assertion? 
I suggest replacing it with a simple `assert` + log message so we won't crush (here) on production, WDYT?